packages/table/src/Row/InternalRowWithRT.tsx

```diff
@@ -77,17 +78,19 @@ const InternalRowWithRT = <T extends LGRowData>({
 export const MemoizedInternalRowWithRT = React.memo(
   InternalRowWithRT,
   (prevProps, nextProps) => {
-    const { children: prevChildren, ...restPrevProps } = prevProps;
-    const { children: nextChildren, ...restNextProps } = nextProps;
+    // Early exit if rows should not be memoized
+    if (!nextProps.shouldMemoizeRows) return false;
 
-    // This allows us to rerender if the child(cell) count changes. E.g. column visibility changes
-    const prevChildrenCount = React.Children.count(prevChildren);
-    const nextChildrenCount = React.Children.count(nextChildren);
+    // Extract children and compare counts
+    const prevChildrenCount = React.Children.count(prevProps.children);
+    const nextChildrenCount = React.Children.count(nextProps.children);
 
-    const propsAreEqual =
-      isEqual(restPrevProps, restNextProps) &&
-      prevChildrenCount === nextChildrenCount;
+    if (prevChildrenCount !== nextChildrenCount) return false;
 
-    return propsAreEqual;
+    // Compare all props except children and shouldMemoizeRows
+    const { children: _1, shouldMemoizeRows: _2, ...prevRest } = prevProps;
+    const { children: _3, shouldMemoizeRows: _4, ...nextRest } = nextProps;
```

Comment:
Just a nit, but if we're omitting more than one prop, I'd use `_.omit`

```suggestion
    const prevRest = omit(prevProps, ['children', 'shouldMemoizeRows']);
    const nextRest = omit(nextProps, ['children', 'shouldMemoizeRows']);
```